src/librustdoc/clean/mod.rs

```diff
@@ -2568,8 +2568,12 @@ fn resolve_type(cx: &DocContext,
     debug!("resolve_type({:?},{:?})", path, id);
     let tcx = match cx.tcx_opt() {
         Some(tcx) => tcx,
-        // If we're extracting tests, this return value doesn't matter.
-        None => return Primitive(Bool),
+        // If we're extracting tests, this return value's accuracy is not important, all we
+        // want is a string representation to help people figure out what doctests are failing.
+        None => {
+            let did = DefId::local(DefIndex::from_u32(0));
+            return ResolvedPath { path: path, typarams: None, did: did, is_generic: false }
```

Comment:
I think it's been awhile since I've poked around this, but could you elaborate a bit on why swapping one dummy type for another was needed to get the names to work out?

Also, could you wrap this to 80 chars? (style of rustdoc)